src/bthread/remote_task_queue.h

```diff
@@ -42,28 +42,35 @@ class RemoteQueue {
     }
 
     bool pop(bthread_t *task) {
-        if (_task_cnt.load(std::memory_order_acquire) > 0 && _tasks.try_dequeue(*task)) {
-            _task_cnt.fetch_sub(1, std::memory_order_release);
+        if (!_tasks.is_empty() && _tasks.try_dequeue(*task)) {
```

Comment:
Should only use ``try_dequeue``. ``is_empty`` is not O(1) and is as expensive as ``try_dequeue``.